### PR TITLE
Change protobuf trait we use

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: gears/src/commands/client/tx.rs [hunk 1]
```diff
@@ -35,10 +34,7 @@ pub struct ClientTxContext {
 }
 
 impl ClientTxContext {
-    pub fn query<
-        Response: Protobuf<Raw> + std::convert::TryFrom<Raw>,
```
Comment (NYBACHOK) [inline gears/src/commands/client/tx.rs:39]: I removed `Protobuf` bound not only 'cause it's makes no sense for us, but rust don't happy with trait bounds for Error.

File: tendermint/Cargo.toml [hunk 2]
```diff
@@ -29,6 +29,7 @@ rand = { workspace = true }
 serde = { workspace = true }
 serde_json = { workspace = true }
 prost = { workspace = true }
+ibc-proto = { workspace = true } # We only need Protobuf trait...
```
Comment (NYBACHOK) [inline tendermint/Cargo.toml:32]: Maybe it makes sense to create our copy of `Protobuf` trait?

File: tendermint/src/types/time/timestamp.rs [hunk 3]
```diff
@@ -336,7 +335,8 @@ impl From<Timestamp> for inner::Timestamp {
     }
 }
 
-impl Protobuf<Timestamp> for Timestamp {}
+impl tendermint_proto::Protobuf<Timestamp> for Timestamp {}
```
Comment (dmvict) [inline tendermint/src/types/time/timestamp.rs:338]: Do we need both versions?

File: x/auth/src/keeper.rs [hunk 4]
```diff
@@ -83,7 +82,7 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, M: Module> AuthKeeper<SK, M> for Keep
         let mut auth_store = ctx.kv_store_mut(&self.store_key);
         let key = create_auth_store_key(acct.get_address().to_owned());
 
-        auth_store.set(key, acct.encode_vec().expect(IBC_ENCODE_UNWRAP))?; // TODO:IBC
+        auth_store.set(key, acct.encode_vec())?;
```
Comment (dmvict) [inline x/auth/src/keeper.rs:85]: Why here we propagate error?

Comment (NYBACHOK) [reply x/auth/src/keeper.rs:85]: Check code again. It was before me

**Review by dmvict — APPROVED** [review]

File: tendermint/src/types/time/timestamp.rs (hunk 3 above)
Comment (joneskm) [reply tendermint/src/types/time/timestamp.rs:338]: I think we might need the tendermint variant inside our tendermint crate

**Pull request merged**